packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -171,113 +173,123 @@ class WebAssetServer implements AssetReader {
     bool testMode = false,
     DwdsLauncher dwdsLauncher = Dwds.start,
   }) async {
-    try {
-      InternetAddress address;
-      if (hostname == 'any') {
-        address = InternetAddress.anyIPv4;
-      } else {
-        address = (await InternetAddress.lookup(hostname)).first;
+    InternetAddress address;
+    if (hostname == 'any') {
+      address = InternetAddress.anyIPv4;
+    } else {
+      address = (await InternetAddress.lookup(hostname)).first;
+    }
+    HttpServer httpServer;
+    dynamic lastError;
+    for (int i = 0; i < 5; i += 1) {
+      try {
+        httpServer = await HttpServer.bind(address, port ?? await globals.os.findFreePort());
+        break;
+      } on SocketException catch (error) {
+        lastError = error;
```

Comment:
Maybe insert `await Future.delayed(Duration(milliseconds: 100))` or something? In case this is happening while something else is struggling to shut down?

Reply:
Done